collect_app/src/main/java/org/odk/collect/android/tasks/DiskSyncTask.java

```diff
@@ -136,11 +137,17 @@ protected String doInBackground(Void... params) {
                                 uriToUpdate.add(new UriFile(updateUri, sqlFile));
                             }
                         } else {
+                            // remove it from the database also
+                            idsToDelete.add(mCursor.getString(
+                                    mCursor.getColumnIndex(FormsColumns._ID)));
                             Log.w(t, "[" + instance
-                                    + "] file referenced by content provider does not exist "
+                                    + "] file referenced by content provider does not exist, hence removed "
                                     + sqlFile);
                         }
                     }
+                    // remove the list from the database
+                    mFormsDao.deleteFormsForFormId(idsToDelete.toArray(new String[idsToDelete.size()]));
```

Comment:
Maybe change the signature of the delete to accept list of String? This is a maybe.